### ISSUE TITLE
Request: make the prompt include a help string by default

### ISSUE THREAD

**Issue by mattdm** [gwsw/less#72]
We're trying to make the experience of new users a little less frightening in Fedora. One situation people get in is finding themselves in an text editor without knowing how to quit — or a pager like Less. I'd like to suggest that the prompt include a string like `(press h for help or q to quit)`.

If there's aversion to making this the overall default, perhaps it could be added to `-M` only, and we could set `LESS=-M` by default in Fedora.

Basically, make the -M string:

    ?f%f .?n?m(%T %i of %m) ..?ltlines %lt-%lb?L/%L. :byte %bB?s/%s. .?e(END) ?x- Next\: %x.:?pB%pB\%..%t (h for help or q to quit)

**Comment by gwsw** [comment]
You can set the prompt to anything you like by setting the -P option.  Can you set `LESS=-PsYOURPROMPT` in the environment?

**Comment by mattdm** [comment]
We could, but

1. I thought it might be nice to have upstream as a default, so it benefits more users
2. Ideally, it'd be the default even in non-M, so nothing fancy needs to be done.
3. If you don't want that, still `LESS=-M` is less mind-blowing to look at than `LESS='-MPM?f%f .?n?m(%T %i of %m) ..?ltlines %lt-%lb?L/%L. :byte
%bB?s/%s. .?e(END) ?x- Next\: %x.:?pB%pB\%..%t (h for help or q to quit)'`
4. That's not just aesthetics -- practically speaking, if the change is to the default, it means that anyone who wants to set other options like `-X` doesn't have to make sure they don't mess up the rest of the obscure incantation.

**Comment by JohnMH** [comment]
Please to not make this change in Fedora. 'less' has exactly what it sounds like, less clutter on your screen so you can actually use your pager.

**Comment by mattdm** [comment]
@JohnMH I don't think @gwsw is the right person to make that case to. :) But, regardless, if that is a concern, that's why I suggested making this the default for `-M` (`--LONG-PROMPT`) as one possible approach.

**Comment by gwsw** [comment]
I'm not really convinced by the arguments against a custom LESS variable. You can certainly sequester the complicated syntax in one variable that no one needs to touch, like
```
OUR_LESS_PROMPT='-M?f%f .?n?m(%T %i of %m) ..?ltlines %lt-%lb?L/%L. :byte %bB?s/%s. .?e(END) ?x- Next\: %x.:?pB%pB\%..%t (h for help or q to quit)'
# 
LESS=-E $OUR_LESS_PROMPT
```

The definition of OUR_LESS_PROMPT could be in /etc/bashrc or whatever so most people won't even see it.

**Comment by JohnMH** [comment]
On Wednesday, November 18, 2020 5:20:56 PM MST gwsw wrote:
> I'm not really convinced by the arguments against a custom LESS variable.
> You can certainly sequester the complicated syntax in one variable that no
> one needs to touch, like ```
> OUR_LESS_PROMPT='-M?f%f .?n?m(%T %i of %m) ..?ltlines %lt-%lb?L/%L. :byte
> %bB?s/%s. .?e(END) ?x- Next\: %x.:?pB%pB\%..%t (h for help or q to quit)' #
> LESS=-E $OUR_LESS_PROMPT
> ```
> 
> The definition of OUR_LESS_PROMPT could be in /etc/bashrc or whatever so
> most people won't even see it.

If this sort of thing must be done, that's definitely the best way to handle 
it.

-- 
John M. Harris, Jr.
Splentity
